src/crowdin_gui.cpp

```diff
@@ -496,33 +484,23 @@ class CrowdinOpenDialog : public wxDialog
             .catch_all(m_activity->HandleError);
     }
 
-    bool IsFileSupported(const wxString& name) const
+    wxString CreateLocalFilename(const long fileId, const wxString& name, const Language& lang, const long projectId, const wxString& projectName)
     {
-        return boost::ends_with(name, ".po") || boost::ends_with(name, ".pot");
-    }
-
-    wxString CreateLocalFilename(const wxString& name, const Language& lang)
-    {
-        wxString cache;
-    #if defined(__WXOSX__)
-        cache = wxGetHomeDir() + "/Library/Caches/net.poedit.Poedit";
-    #elif defined(__UNIX__)
-        if (!wxGetEnv("XDG_CACHE_HOME", &cache))
-            cache = wxGetHomeDir() + "/.cache";
-        cache += "/poedit";
-    #else
-        cache = wxStandardPaths::Get().GetUserDataDir() + wxFILE_SEP_PATH + "Cache";
-    #endif
-
-        cache += wxFILE_SEP_PATH;
-        cache += "Crowdin";
-
-        if (!wxFileName::DirExists(cache))
-            wxFileName::Mkdir(cache, wxS_DIR_DEFAULT, wxPATH_MKDIR_FULL);
-
-        auto basename = name.AfterLast('/').BeforeLast('.');
-
-        return wxString::Format("%s%c%s_%s_%s.po", cache, wxFILE_SEP_PATH, m_info.name, basename, lang.Code());
+        wxFileName crowdinFileName(name, wxPATH_UNIX),
+                   localFileName =
+            CloudSyncDestination::GetCacheDir() + wxFILE_SEP_PATH + "Crowdin"
+            + wxFILE_SEP_PATH + projectName
+            + wxFILE_SEP_PATH + lang.Code()
+            + wxFILE_SEP_PATH + crowdinFileName.GetPath() + wxFILE_SEP_PATH
+            << projectId << '_' << fileId << '_' << crowdinFileName.GetFullName();
+       
+        if(localFileName.GetExt().CmpNoCase("po"))// if not PO
+            localFileName.SetFullName(localFileName.GetFullName() + ".xliff");
```

Comment:
So, storing XLIFF files as `foo.xliff.xliff`?

Reply:
Yep, since we need to know remote (Crowdin side) format/extension of locally opened `Xliff` file to provide correct extension here https://github.com/tetatetit/poedit/blob/cc589293a710f0f0ecfaed767de94f3fe829edd0/src/crowdin_client.cpp#L400 otherwise error response received. And at least me was not able to find a better solution than just to add `.xliff` extension to remote filename with included it remote extension as well when saving locally. Of course I considered just _replacing_ remote extension with `.xliff` but how remote e.g. `some_file_with_same_base_name` and `some_file_with_same_base_name.xliff` would be stored locally in that case without collision while both would become  `some_file_with_same_base_name.xliff` with such approach? And how to distinguish between their remote extensions/formats if opened locally and then uploaded (while correct extension should be provided as mentioned above). That is why it happens that remote `some_file_with_same_base_name.xliff` will be saved as `some_file_with_same_base_name.xliff.xliff` locally while `some_file_with_same_base_name` as `some_file_with_same_base_name.xliff`. I admit it's kind of hack but if you have any other ideas - I would highly appreciate for sharing them here. Many thanks.

P. S. For `.po` files everything is much simple - since remote `.po` is always local `.po`